File: client/lua/krpc/test/test_enum.lua
local luaunit = require 'luaunit'
local class = require 'pl.class'
local schema = require 'krpc.test.Test'

local TestEnum = class()

function TestEnum:test_enums()
  luaunit.assertEquals(schema.TestEnum.a, 0)
  luaunit.assertEquals(schema.TestEnum.b, 1)
  luaunit.assertEquals(schema.TestEnum.c, 2)
end

return TestEnum

File: client/python/krpc/test/test_enum.py
import unittest
import krpc.test.schema.Test as TestSchema

class TestEnum(unittest.TestCase):

    def test_enums(self):
        self.assertEqual(0, TestSchema.a)
        self.assertEqual(1, TestSchema.b)
        self.assertEqual(2, TestSchema.c)

if __name__ == '__main__':
    unittest.main()
